transport-native-unix-common/src/main/c/netty_unix_socket.c

```diff
@@ -755,18 +755,16 @@ static void netty_unix_socket_setSoLinger(JNIEnv* env, jclass clazz, jint fd, ji
 }
 
 static void netty_unix_socket_setTrafficClass(JNIEnv* env, jclass clazz, jint fd, jint optval) {
-    /* Try to set the ipv6 equivalent, but don't throw if this is an ipv4 only socket. */
-    int rc = netty_unix_socket_setOption0(fd, IPPROTO_IPV6, IPV6_TCLASS, &optval, sizeof(optval));
-    if (rc < 0 && errno != ENOPROTOOPT) {
-        netty_unix_socket_setOptionHandleError(env, errno);
-    }
+    if (socketType == AF_INET6) {
+        netty_unix_socket_setOption(env, fd, IPPROTO_IPV6, IPV6_TCLASS, &optval, sizeof(optval));
```

Comment:
I'm concerned that this change deletes *all* error handling for setTrafficClass().  The simplest approach for the `AF_INET6` branch is to try both, and throw an error if both fail, e.g.:

    const int rc6 = netty_unix_socket_setOption(env, fd, IPPROTO_IPV6, IPV6_TCLASS, &optval, sizeof(optval));
    const int rc6_errno = rc6 < 0 ? errno : 0;
    const int rc4 = netty_unix_socket_setOption(env, fd, IPPROTO_IP, IP_TOS, &optval, sizeof(optval));
    const int rc4_errno = rc4 < 0 ? errno : 0;
    if (rc6 < 0 && rc4 < 0) {
        // some logic to figure out which errno is more relevant.
    }

This removes the `__linux__` special case, assuming that other platforms provide a reasonable errno.

Reply:
@pmarks-net maybe I am missing something but as far as i can tell it does correctly handle errors. Please note we call `netty_unix_socket_setOption` and NOT `netty_unix_socket_setOption0`.